### ISSUE TITLE
New transformers module `4.37.2` saving the fine-tuned model as model.safetensors

### ISSUE THREAD

**Issue by pratikchhapolika** [huggingface/transformers#29365]
### System Info

I recently encountered an unexpected behavior while using the latest `Transformers package version 4.37.2.` 
Tf2.9, Python3.9

Upon training a model and attempting to save it using the `Trainer module`, instead of the expected `pytorch_model.bin` file, the model is being saved as `model.safetensors`. 

This departure from the usual saving convention has raised questions regarding whether this is a **new feature** introduced in the Transformers package or a feature of the underlying PyTorch package.

Questions:

Is the saving behavior observed (model.safetensors instead of pytorch_model.bin) a newly introduced feature in the Transformers package or a characteristic of the PyTorch package?

How can I ensure that both `model.safetensors and pytorch_model.bin` are saved concurrently while using the Trainer module for model fine-tuning and saving?

### Who can help?

@ArthurZucker @sgugger 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

training_args = TrainingArguments(
        output_dir="v4_1",  # output directory
        num_train_epochs=10,  # total number of training epochs
        per_device_train_batch_size=8,  # batch size per device during training
        per_device_eval_batch_size=8,  # batch size for evaluation
        warmup_steps=(len(X_train)*6)//80,  # number of warmup steps for learning rate scheduler
        weight_decay=0.1,  # strength of weight decay
        logging_steps=500,
        evaluation_strategy='epoch',
        save_strategy = "epoch",
        logging_dir="logs",
        do_eval=  True,
        fp16=True,
        load_best_model_at_end = True,
        dataloader_num_workers = 4,
        #metric_for_best_model = "eval_precision",
        greater_is_better = "False",
        learning_rate = 1e-5
    )

trainer = Trainer(
        model=model,  # the instantiated 🤗 Transformers model to be trained
        args=training_args,  # training arguments, defined above
        train_dataset=train_dataset,  # training dataset
        eval_dataset=val_dataset, # evaluation dataset
        compute_metrics=compute_metrics # function to compute metrics for model eval
    )

trainer.train()
trainer.save_model("model_new")

### Expected behavior

save pytorch_model.bin

**Comment by ArthurZucker** [comment]
Hey! This is part of our big refactoring that makes sure we save our models as `safetensors` vs `.bin` files which can execute arbitrary code. 

Inviting you to read [the doc](https://huggingface.co/docs/safetensors/en/index) here and [the corresponding argument to `save_pretrained`](https://huggingface.co/docs/transformers/main_classes/model#transformers.PreTrainedModel.save_pretrained.safe_serialization).

TLDR; it's expected

**Comment by pratikchhapolika** [comment]
> the corresponding argument to `save_pretrained`

Perfect!! Thank you for replying! Answers my Question.
